salt/modules/datadog_api.py

```diff
@@ -262,3 +259,426 @@ def post_event(api_key=None,
     else:
         ret['comment'] = 'Error in posting event.'
     return ret
+
+
+# The monitor functions simply pass tags and options along to the datadog
+# library. No modifications are made.
+# pylint: disable=dangerous-default-value
+def create_monitor(api_key=None,
+                   app_key=None,
+                   type=None,
+                   query=None,
+                   name=None,
+                   message=None,
+                   tags=[],
```

Comment:
I see your comment above, but we need to only use `None` as defaults here. Then you can handle creating lists within the function if `None`.